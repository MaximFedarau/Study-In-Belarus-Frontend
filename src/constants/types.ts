//Interface for Drawer Link Component
interface DrawerLinkInterface {
  link: string;
  name: string;
}

//Inteface for NAVIGATION_DRAWER_ITEMS
export interface NavigatonDrawerItemsInterface {
  [key: string]: DrawerLinkInterface[];
}

//Interface for News
export interface NewsItemInterface {
  title: string;
  date?: Date;
  image: string;
}
