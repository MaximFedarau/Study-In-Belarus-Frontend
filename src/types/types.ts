import React from 'react';
import { DEFAULT_SIZES } from './enum';

// Interface for Styles in Media Queries (CSS) Utils
export type MEDIA_QUERIES_STYLES = React.CSSProperties | { [key: string]: any };

//Interface for Drawer Link Component
interface DrawerLinkInterface {
  link: string;
  name: string;
}

//Inteface for NAVIGATION_DRAWER_ITEMS
export interface NavigatonDrawerItemsInterface {
  [key: string]: DrawerLinkInterface[];
}

//Interface for Home page Bubbles
export type HomeBubblesStylesInterface = {
  [key in DEFAULT_SIZES]: {
    container: React.CSSProperties;
    mainText: React.CSSProperties;
    subtext: React.CSSProperties;
  };
};

//Interface for Carousel Student Interface
export interface CarouselStudentInterface {
  name: string;
  speech: string;
  image: string;
  country: string;
  education: string;
}

//Interface for News
export interface NewsItemInterface {
  title: string;
  date?: Date;
  image: string;
}
