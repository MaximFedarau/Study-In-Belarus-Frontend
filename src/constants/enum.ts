export enum PAGES_TITLES {
  HOME = 'Home',
  ADVANTAGES_OF_STUDYING_IN_BELARUS = 'Advantages of Studying in Belarus',
  NOT_FOUND = '404',
}

//Button Styling Variant
export enum BUTTON_VARIANT {
  CONTAINED = 'contained',
  OUTLINED = 'outlined',
}

//Default Sizes System
export enum DEFAULT_SIZES {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
}
