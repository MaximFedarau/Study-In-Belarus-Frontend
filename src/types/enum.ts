// Styling sizes for media queries
export enum STYLING_SIZES {
  LARGE_TABLET = '1170px', // about 12 times per project
  SMALL_TABLET = '1070px', // about 9 times per project
  LARGE_PHONE = '800px', // about 12 times per project
  SMALL_PHONE = '400px', // about 1 time per project
}

export enum PAGES_TITLES {
  HOME = 'Home',
  ADVANTAGES_OF_STUDYING_IN_BELARUS = 'Advantages of Studying in Belarus',
  NOT_FOUND = '404',
}

//Button Styling Variant
export enum BUTTON_VARIANT {
  CONTAINED = 'contained',
  OUTLINED = 'outlined',
}

//Default Sizes System
export enum DEFAULT_SIZES {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
}
