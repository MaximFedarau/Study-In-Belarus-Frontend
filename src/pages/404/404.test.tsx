//Types
import { PAGES_TITLES } from 'types/enum';

//RTL
import { render } from '@testing-library/react';

//Page
import Page404 from 'pages/404/404.page';

describe('404 page attributes working correcty.', () => {
  test('404 page title is correct.', () => {
    console.log(document);
    expect(document.title).toBe(PAGES_TITLES.NOT_FOUND);
  });
});

// ! Warning. I added snapshot checking here, because I don't want the page to ever change much.
describe('404 page UI is correct.', () => {
  const page = render(<Page404 />);
  expect(page).toMatchSnapshot();
});
