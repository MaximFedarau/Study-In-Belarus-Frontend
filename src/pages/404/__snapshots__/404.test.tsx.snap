// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-n080w8"
      >
        <div
          class="MuiBox-root css-1avs8wp"
        >
          <header
            class="css-19dhr21"
          >
            Study in Belarus
          </header>
          <section
            class="css-93ppzn"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ptg5so-MuiTypography-root"
            >
              A 404 error has occurred. To continue:
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ptg5so-MuiTypography-root"
            >
              Reload current page, or
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ptg5so-MuiTypography-root"
            >
              Navigate to the home page. Also check your internet connection and reload current page.
            </p>
          </section>
          <footer
            class="css-zofoza"
          >
            © 2022 Study in Belarus. All rights reserved.
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-n080w8"
    >
      <div
        class="MuiBox-root css-1avs8wp"
      >
        <header
          class="css-19dhr21"
        >
          Study in Belarus
        </header>
        <section
          class="css-93ppzn"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ptg5so-MuiTypography-root"
          >
            A 404 error has occurred. To continue:
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ptg5so-MuiTypography-root"
          >
            Reload current page, or
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ptg5so-MuiTypography-root"
          >
            Navigate to the home page. Also check your internet connection and reload current page.
          </p>
        </section>
        <footer
          class="css-zofoza"
        >
          © 2022 Study in Belarus. All rights reserved.
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
